packages/plugin-ext/src/main/browser/notebooks/renderers/output-webview-internal.ts

```diff
@@ -160,8 +160,8 @@ export async function outputWebviewPreload(ctx: PreloadContext): Promise<void> {
             if (this.rendererApi) {
                 return this.rendererApi;
             }
-
-            const rendererModule = await __import(this.data.entrypoint.uri) as { activate: rendererApi.ActivationFunction };
+            const baseUri = window.location.href.replace(/\/webview\/index\.html.*/, '');
```

Comment:
The `/webview/` part used here is part of an implementation detail of the [`WebviewEnvironment`](https://github.com/eclipse-theia/theia/blob/f1056b3c567e8cc9d3807850164d4f48967f28a6/packages/plugin-ext/src/main/browser/webview/webview-environment.ts#L55-L61), making it fragile in case adopters decide to change the path.

Instead I'd advocate for a different approach. Why don't we pass the full `URI` to the `data.entrypoint.uri` property? Currently, we only pass the path of the renderer, but we should probably pass the full URI.

Reply:
but if we pass the full path wouldn't we get cross origin issues?

Reply:
I just confirmed we get cross origin errors if we pass in the full uri. 
What we could do would be to pass the full path and then just use the host and port from the webview url

Reply:
I guess, yeah we do...

Ok, I'll approve for now, and we can get back to this in case someone actually has trouble with this approach. I assume no one actually manipulates the `/webview` path anyway.